src/fpsCounter.ts

```diff
@@ -0,0 +1,27 @@
+export class FpsCounter {
+    private readonly SECOND = 1000;
+    private static _fpsCounter: FpsCounter;
+
+    private fpsLabel: HTMLElement;
+    private recentDraws: number[] = [];
+
+    public static getFpsCounter(): FpsCounter {
+        return this._fpsCounter || (this._fpsCounter = new FpsCounter());
+    }
+
+    private constructor() {
+        this.fpsLabel = document.getElementById("fps-status")!;
+    }
+
+    public addDraw(): void {
```

Comment:
Can this be renamed to 'countFrame' or similar? 'addDraw' is just a little too unclear to me.

Reply:
Had thought about merging the two functions together, since I feel like they are will almost always be used together. Could also create a third function that calls them both, but feel a bit much.